pkg/kubelet/apis/config/types.go

```diff
@@ -656,3 +661,16 @@ type MemorySwapConfiguration struct {
 	// +optional
 	SwapBehavior string
 }
+
+// RateLimitConfiguration is used for setting the rate limits for the kubelet APIs exposed
+// locally on the node as gRPC over UNIX-domain sockets, like the podresources API
+type RateLimitConfiguration struct {
+	// MaxFrequency  defines the maximum allowed frequency of the calls represented
+	// as number of calls per second, cumulating all the calls.
+	// Omitting the value or setting to zero means no limit.
+	MaxFrequency float64
```

Comment:
what is the behavior when this limit is hit? does the last fetched result get replayed or does the caller get an error? if replayed, how is that indicated? if an error, how is the client expected to react, and what are the implications of clients being able to starve/disrupt each other?